code/datums/wires/robot.dm

```diff
@@ -59,8 +63,12 @@
 	switch(wire)
 		if(WIRE_AI) // Cut the AI wire to reset AI control.
 			if(!mend)
+				if(R.shell)
+					R.undeploy()
+					R.notify_ai(AI_SHELL)

```

Comment:
Have you tested these? It looks like the code would tell the AI that a new borg/shell has been connected when you are in fact disconnecting it. If you want to include messages for borgs disconnected by cutting this wire, you need to make a new message or modify one to indicate the event correct.